### ISSUE TITLE
Rejection reasons

### ISSUE THREAD

**Issue by ghost** [tuupola/slim-jwt-auth#53]
Tried to collect possible rejection reasons to react on malformed, expired or missing tokens to use in client for proper error handling.

Got these from exceptions declared in the lib itself, are there more?

1. Token not found
2. Expired token
3. Signature verification failed

**Comment by tuupola** [comment]
Underlying library `firebase/php-jwt` seems to have these: 

https://github.com/firebase/php-jwt/blob/master/src/JWT.php#L58-L62

**Comment by ghost** [comment]
Yes thanks a lot. Would be nice to have those listed in the docu as well.

**Comment by tuupola** [comment]
Yeah, documentation is always lacking... Will keep this in mind. Thanks for the heads up!
